geniza/corpus/models.py

```diff
@@ -1362,9 +1362,38 @@ def merge_with(self, merge_docs, rationale, user=None):
     def _merge_footnotes(self, doc):
         # combine footnotes; footnote logic for merge_with
         for footnote in doc.footnotes.all():
-            # check for match; add new footnote if not a match
+            # check for match
             equiv_fn = self.footnotes.includes_footnote(footnote)
-            if not equiv_fn:
+            if footnote.annotation_set.exists():
+                # if the footnote has annotations, reassign them to matching/created footnote
+                (self_fn, _) = self.footnotes.get_or_create(
+                    # for multiselect field list, need to cast to list to compare
+                    doc_relation__in=list(footnote.doc_relation),
+                    source_id=footnote.source.pk,
+                    # TODO: What to do if found, but location or notes don't match? Discard?
+                    defaults={
+                        "notes": footnote.notes,
+                        "location": footnote.location,
+                        "url": footnote.url,
+                    },
+                )
+                # reassign each annotation's footnote to the footnote on the merge result doc
+                for annotation in footnote.annotation_set.all():
+                    annotation.footnote = self_fn
+                    annotation.save()
```

Comment:
Is the problem that `includes_footnote` comparison doesn't account for annotations?  Should it include that when checking, or should it be optionally included? (... trying to think when you would not want to include annotations when comparing footnotes, but it could be an option you specify when you call the includes method if we think we don't always need it).

Alternately, would it make sense to move annotations to the equivalent footnote if we don't want to include annotations in the footnote equality check?

I'm not totally following this logic here but I think it might be more complex than needed / not in the right place. Seems like the simplest solution would just to include annotations in the footnote equivalence check, but I may be misunderstanding the problem.

Reply:
ok, made the mistake of commenting before reading through everything - I see you've made some updates to the includes footnote method and are explicitly declaring footnotes with annotations to not be equivalent (which seems reasonable). 

I think you should add a comment and make this more explicit: these are two different cases where there's no equivalent footnote. If I'm understanding correctly, the options are to find an appropriate footnote and re-associate the annotations or create a new footnote. IMO notes and locations should be preserved, because they may include important information about where the content is located or if the edition has been revised by someone.

I still think parameterizing the equiv fn check might be helpful and simplify this code; is it helpful to think in terms of exactly equivalent footnotes (your updated logic) or nearly equivalent ( ~ old logic)?